src_c/rwobject.c

```diff
@@ -722,22 +737,43 @@ _rwops_from_pystr(PyObject *obj)
     if (obj != NULL) {
         SDL_RWops *rw = NULL;
         PyObject *oencoded;
+        char* encoded = NULL;
+        char* ext = NULL;
+        char* extension = NULL;
+
+        //longest extension is 4 chars, plus NULL at end
+        extension = malloc(sizeof(char)*5);
+        if (extension == NULL) {
+            return (SDL_RWops*)PyErr_NoMemory();
+        }
+
         oencoded = pg_EncodeString(obj, "UTF-8", NULL, NULL);
         if (oencoded == NULL) {
             return NULL;
         }
         if (oencoded != Py_None) {
-            rw = SDL_RWFromFile(Bytes_AS_STRING(oencoded), "rb");
+            encoded = Bytes_AS_STRING(oencoded);
+            rw = SDL_RWFromFile(encoded, "rb");
+            ext = strrchr(encoded, '.');
+            if (ext) {
+                ext++;
+                strcpy(extension, ext);
```

Comment:
Perhaps move the malloc to something like this down here (untested):
```c
        if (ext && strlen(ext)) {
            extension = malloc(strlen(ext));
            if (extension == NULL) {
                return (SDL_RWops*)PyErr_NoMemory();
            }
            strcpy(extension, ext);
        }
```

Reply:
This kind of c code still makes me extremely cautious.

I've gone with this:
```c
ext = strrchr(encoded, '.');        
if (ext && strlen(ext) > 1) {
    ext++;
    extension = malloc(strlen(ext)+1);
    if (extension == NULL) {
        return (SDL_RWops*)PyErr_NoMemory();
    }
    strcpy(extension, ext);
}
```

strlen() > 1 because of the case where the filename ends with a period
malloc with a + 1 to leave space for the null terminator of the string

Reply:
That's looking good now :)